source/solvers/postprocessing_velocities.cc

```diff
@@ -0,0 +1,61 @@
+#include <solvers/postprocessing_velocities.h>
+
+
+template <int dim, typename VectorType, typename DofsType>
+void
+AverageVelocities<dim, VectorType, DofsType>::calculate_average_velocities(
+  const VectorType &                        local_evaluation_point,
+  const std::shared_ptr<SimulationControl> &simulation_control,
+  const Parameters::PostProcessing &        post_processing,
+  const DofsType &                          locally_owned_dofs,
+  const MPI_Comm &                          mpi_communicator)
+{
+  double epsilon = 1e-6;
+  double total_time =
+    simulation_control->get_current_time() - post_processing.initial_time;
+  double dt = simulation_control->get_time_step();
+
+  // When averaging velocities begins (time = initial time)
+  if (abs(total_time) < epsilon)
```

Comment:
This is not a safe check. What happens if initial_time=10 and your time step brings you to 10.01. This check will fail. You inherently assume that you will start averaging "pile poil" at a multiple of your time step. This is impossible with adaptative time-stepping. You need to find a more robust mechanism.

Reply:
Oh ok, did not think about it, I'm not very familiar with the adaptative time-stepping.